### PR TITLE
make build/common.sh use docker build --pull=true FROM k8s.gcr.io/kube-cross

### LINKED ISSUE #79048: build/run.sh failure due to fetch false preventing from being pulled
**What happened**:
I'm noob just trying [to follow the README](https://github.com/kubernetes/kubernetes#you-have-a-working-docker-environment) for the first time, but `build/run.sh` fails early on:

```
$ build/run.sh make kubectl
+++ [0614 23:18:26] Verifying Prerequisites....
+++ [0614 23:18:26] Building Docker image kube-build:build-8ef9608e83-5-v1.12.5-1
+++ Docker build command failed for kube-build:build-8ef9608e83-5-v1.12.5-1

Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
STEP 1: FROM k8s.gcr.io/kube-cross:v1.12.5-1
Error: error creating build container: no such image "containers-storage:[overlay@/home/vorburger/.local/share/containers/storage+/tmp/1000:overlay.mount_program=/usr/bin/fuse-overlayfs]k8s.gcr.io/kube-cross:v1.12.5-1"

To retry manually, run:

docker build -t kube-build:build-8ef9608e83-5-v1.12.5-1 --pull=false /home/vorburger/Kubernetes/kubernetes/_output/images/kube-build:build-8ef9608e83-5-v1.12.5-1

!!! [0614 23:18:27] Call tree:
!!! [0614 23:18:27]  1: build/run.sh:31 kube::build::build_image(...)
!!! Error in build/../build/common.sh:427
  Error in build/../build/common.sh:427. 'return 1' exited with status 1
Call stack:
  1: build/../build/common.sh:427 kube::build::build_image(...)
  2: build/run.sh:31 main(...)
Exiting with status 1

```

**What you expected to happen**:
I guess i expected `build/run.sh` as advertised in the README? :smile: 

**How to reproduce it (as minimally and precisely as possible)**:
1. Fedora 30 Server
1. sudo dnf install podman-docker
1. docker rmi k8s.gcr.io/kube-cross
1. git clone https://github.com/kubernetes/kubernetes.git ; cd kubernetes
1. `build/run.sh make kubectl`

### PR DESCRIPTION
**Pull request by vorburger**
/kind bug

Fixes #79048

```release-note
NONE
```

### CODE REVIEW

File: build/common.sh [hunk 1]
```diff
@@ -424,7 +424,7 @@ function kube::build::build_image() {
 
   kube::build::update_dockerfile
   kube::build::set_proxy
-  kube::build::docker_build "${KUBE_BUILD_IMAGE}" "${LOCAL_OUTPUT_BUILD_CONTEXT}" 'false'
+  kube::build::docker_build "${KUBE_BUILD_IMAGE}" "${LOCAL_OUTPUT_BUILD_CONTEXT}" 'true'
```
Comment (BenTheElder) [inline build/common.sh]: I assume we're not trying to pull every time, perhaps something else should ensure it exists as opposed to just pulling every time.

Have we analyzed why this was previously set to true? git history?

Comment (vorburger) [reply build/common.sh]: > I assume we're not trying to pull every time, perhaps something else should ensure it exists as opposed to just pulling every time.

I see, but in that case the doc is wrong... if someone can enlighten me what pre-req step I may be missing, I'd be delighted. OTH seeing `FROM k8s.gcr.io/kube-cross:v1.12.5-1` in the respective `Dockerfile` does make it seem to me this was intended to pull 

> Have we analyzed why this was previously set to true? git history?

git blame shows d403c2c16893c98aaeb1ec30bf352b95f4006541 last touched that line, but it just moved a few lines up. Do you have any tricks up your sleeve I should learn how to to easily see the previous revision git blame until you see where it was originally introduced?

**Comment by cblecker** [comment]
Based on https://github.com/kubernetes/kubernetes/issues/79048#issuecomment-502299635, what exists there is correct and works for docker-ce.

/close

**Pull request closed without merging**